Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragmentViewModel.kt

```diff
@@ -338,7 +341,7 @@ class HomeFragmentViewModel @AssistedInject constructor(
             add(
                 CreateVaccinationHomeCard.Item(
                     onClickAction = {
-                        // TODO: implement in another PR
+                        openVaccinationRegistrationFlow.postValue(Unit)
```

Comment:
why do we need a new singleLiveData here? can't we just use `routeToScreen `?

Reply:
yes, this could be implemented with `routeToScreen`

Reply:
actually, I'm updating `startDestination` of `nav_graph` according to if consent has been granted so it's not just `routeToScreen`. See implementation in the `HomeFragment`

Reply:
Ok, we probably need to refactor the navigation logic of HomeFragment and create a sealed class for that instead of using the NavDirections directly to allow us pass data in it